pkg/lib/oauth/handler/handler_token.go

```diff
@@ -682,6 +683,23 @@ func (h *TokenHandler) issueTokensForAuthorizationCode(
 		}
 	}
 
+	// Update auth_time of the offline grant if possible.
+	if sid := code.IDTokenHintSID; sid != "" {
+		if typ, sessionID, ok := oidc.DecodeSID(sid); ok && typ == session.TypeOfflineGrant {
+			offlineGrant, err := h.OfflineGrants.GetOfflineGrant(sessionID)
+			if err == nil {
+				if s.AuthenticatedAt.After(offlineGrant.AuthenticatedAt) {
+					offlineGrant.AuthenticatedAt = s.AuthenticatedAt
+					expiry := oauth.ComputeOfflineGrantExpiryWithClient(offlineGrant, client)
```

Comment:
There is a side effect that the expiry of refresh token will be extended after re-auth, I think it is acceptable?

Reply:
Actually it will not. If we look at `oauth.ComputeOfflineGrantExpiryWithClient`, the factors affecting the expiry is `CreatedAt`, `LastAccessAt` and the client config. We do not modify any of these in the reauthentication explicitly. Of course the last access at will be updated when the access token is used to refresh ID token.